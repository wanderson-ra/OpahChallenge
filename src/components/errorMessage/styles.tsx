import { heightPercentageToDP as height, widthPercentageToDP as width } from "react-native-responsive-screen";

import styled from "styled-components/native";

import { metrics } from "src/configurations/metrics";

export const Container = styled.View`
    flex: 1;
    justify-content: center;
    align-items: center;
`;

export const Message = styled.Text`
    margin-top: ${height(2)}px;
    text-align: center;
    font-size: ${metrics.font.small}px;
    color: ${(props): string => props.theme.font.secondary};
    width: ${width(63)}px;
`;
