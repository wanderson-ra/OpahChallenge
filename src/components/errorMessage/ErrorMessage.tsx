import React from "react";
import { heightPercentageToDP as height, widthPercentageToDP as width } from "react-native-responsive-screen";
import Icon from "react-native-vector-icons/MaterialCommunityIcons";

import { useTheme } from "styled-components/native";

import { Container, Message } from "./styles";

import { Button } from "../button/Button";

export interface MessageProps {
    message: string;
    icon: string;
    buttonTitle: string;
}

interface ErrorMessageProps {
    onPress: () => void;
    messageProps: MessageProps;
}

export const ErrorMessage: React.FC<ErrorMessageProps> = (errorMessageProps) => {
    const theme = useTheme();
    const { messageProps, onPress } = errorMessageProps;

    return (
        <Container>
            <Icon testID={"icon"} name={messageProps.icon} size={height(5)} color={theme.icon.default} />

            <Message testID={"message"}>{messageProps.message}</Message>

            <Button
                containerStyle={{ width: width(50), height: height(4) }}
                enable={true}
                testID={"buttonPageMessage"}
                onPress={(): void => onPress()}
                title={messageProps.buttonTitle}
            />
        </Container>
    );
};
