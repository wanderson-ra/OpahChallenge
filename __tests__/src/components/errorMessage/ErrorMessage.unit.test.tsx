import React from "react";
import "react-native";
import renderer from "react-test-renderer";

import { ThemeProvider } from "styled-components/native";

import colorTheme from "../../../../src/configurations/themes/theme";

import { ErrorMessage, MessageProps } from "../../../../src/components/errorMessage/ErrorMessage";

describe("Test of ErrorMessage", () => {
    it("Render with success", () => {
        const message = "anyMessage";
        const icon = "lan-disconnect";
        const buttonTitle = "anyButtonTitle";
        const onPress = jest.fn();

        const messageProps: MessageProps = {
            message: message,
            icon: icon,
            buttonTitle: buttonTitle,
        };

        const errorMessage: renderer.ReactTestRenderer = renderer.create(
            <ThemeProvider theme={colorTheme.light}>
                <ErrorMessage messageProps={messageProps} onPress={onPress} />
            </ThemeProvider>
        );

        const messageText = errorMessage.root.findByProps({ testID: "message" });
        const iconText = errorMessage.root.findByProps({ testID: "icon" });

        expect(messageText.props.children).toEqual(message);
        expect(iconText.props.name).toEqual(icon);

        const button = errorMessage.root.findByProps({ testID: "buttonPageMessage" });

        renderer.act(() => {
            button.props.onPress();
            expect(onPress).toHaveBeenCalledTimes(1);
        });
    });

    it("Render with error button title", () => {
        const message = "anyMessage";
        const icon = "lan-disconnect";
        const buttonTitle = "anyButtonTitle";
        const onPress = jest.fn();

        const messageProps: MessageProps = {
            message: message,
            icon: icon,
            buttonTitle: buttonTitle,
        };

        const errorMessage: renderer.ReactTestRenderer = renderer.create(
            <ThemeProvider theme={colorTheme.light}>
                <ErrorMessage messageProps={messageProps} onPress={onPress} />
            </ThemeProvider>
        );

        const messageText = errorMessage.root.findByProps({ testID: "message" });
        const iconText = errorMessage.root.findByProps({ testID: "icon" });

        expect(messageText.props.children).toEqual(message);

        expect(iconText.props.name).toEqual(icon);

        const button = errorMessage.root.findByProps({ testID: "buttonPageMessage" });

        renderer.act(() => {
            button.props.onPress();
            expect(onPress).toHaveBeenCalledTimes(1);
        });
    });
});
