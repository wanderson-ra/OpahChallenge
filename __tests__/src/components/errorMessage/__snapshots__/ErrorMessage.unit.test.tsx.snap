// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of ErrorMessage Render with snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": "#757575",
          "fontSize": 66.5,
        },
        undefined,
        Object {
          "fontFamily": "Material Design Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
    testID="icon"
  >
    ó°Œ™
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#757575",
          "fontSize": 28,
          "marginTop": 26.5,
          "textAlign": "center",
          "width": 472.5,
        },
      ]
    }
    testID="message"
  >
    anyMessage
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#880e4f",
        "borderRadius": 13.5,
        "height": 53.5,
        "justifyContent": "center",
        "marginTop": 66.5,
        "opacity": 1,
        "width": 375,
      }
    }
    testID="buttonPageMessage"
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 28,
            "fontWeight": "700",
          },
          undefined,
          Object {
            "opacity": 1,
          },
        ]
      }
      testID="title"
    >
      anyButtonTitle
    </Text>
  </View>
</View>
`;
